pkgs/build-support/fetchurl/default.nix

```diff
@@ -116,13 +116,15 @@ let
     else if sha1   != "" then { outputHashAlgo = "sha1";   outputHash = sha1; }
     else if cacert != null then { outputHashAlgo = "sha256"; outputHash = ""; }
     else throw "fetchurl requires a hash for fixed-output derivation: ${lib.concatStringsSep ", " urls_}";
+
+  nameFromURLStripQuery = url: lib.nameFromURL url "?";
 in
 
 stdenvNoCC.mkDerivation {
   name =
     if showURLs then "urls"
     else if name != "" then name
-    else baseNameOf (toString (builtins.head urls_));
+    else nameFromURLStripQuery (toString (builtins.head urls_));
```

Comment:
```suggestion
    else lib.nameFromURL (toString (builtins.head urls_)) "?";
```

Reply:
In some situations it will end up with a tailing slash, to handle that we may need a seperated function. However, I'm not sure if it is suitable to land in `lib`, or deserve a better name rather than `nameFromURLStripQuery `.